File: 0001_two_sum/two_sum.js
/**
 * Time:    O(n)
 * Space:   O(n)
 * @param   {number[]}  nums
 * @param   {number}    target
 * @return  {number[]}
 */
const twoSum = (nums, target) => {
    const map = {}
    for (let i in nums) {
        const curr = nums[i]
        const compliment = target - curr
        if (map[compliment] !== undefined) {
            return [map[compliment], i]
        }
        map[curr] = i
    }
    return []
}

File: 0001_two_sum/two_sum.py
from typing import List


def two_sum(nums: List[int], target: List[int]) -> list:
    """
    Time: O(n)
    Space: O(n)
    """
    map = {}
    for i, e in enumerate(nums):
        compliment = target - e
        if compliment in map:
            return [map[compliment], i]
        else:
            map[e] = i
    return []
